File: Go/169. Majority Element.go
package main

//goland:noinspection GoUnusedFunction
func majorityElement(nums []int) int {
	count := 1
	majority := nums[0]

	for i := 1; i < len(nums); i++ {
		if count == 0 {
			majority = nums[i]
			count++
		} else if majority == nums[i] {
			count++
		} else {
			count--
		}
	}

	return majority
}

File: PHP/169. Majority Element.php
<?php

/**
 *  @noinspection PhpIllegalPsrClassPathInspection, PhpMultipleClassDeclarationsInspection
 */
class Solution
{
    /**
     * @param  int[]  $nums
     *
     * @noinspection PhpUnused
     */
    public function majorityElement(array $nums): int
    {
        $majority = $nums[0];
        $count = 1;

        for ($i = 1; $i < count($nums); $i++) {
            if ($count === 0) {
                $majority = $nums[$i];
                $count = 1;
            } elseif ($nums[$i] === $majority) {
                $count++;
            } else {
                $count--;
            }
        }

        return $majority;
    }
}
